File: sunny.cpp
#include<iostream>
#include<cmath>
using namespace std;
bool isPerfectSquare(int n)
{
	if (n<0)
	{
		return false;
	}
	int r = int(pow(n , 0.5));
	return r*r == n;
}
bool isSunny(int n)
{
	if(isPerfectSquare(n+1))
	{
		return true;
	}
	return false;
}
int main()
{
	int n;
	cout<<"Enter a number: ";
	cin>>n;
	
	if(isSunny(n))
	{
		cout<<"Is a Sunny Number."<<endl;
	}
	else
	{
		cout<<"Not a Sunny Number."<<endl;
	}
}

File: sunny.py
def isPerfectSquare(n: int) -> bool:
    if n < 0:
        return False
    r = int(n ** 0.5)
    return r * r == n

def isSunny(n:int)->bool:
    if(isPerfectSquare(n+1)):
        return True
    return False

if __name__ == "__main__":
    n = int(input("Enter a number : "))
    if(isSunny(n)):
        print("Is a Sunny Number.")
    else:
        print("Not a Sunny Number.")
